src/main/java/io/cryostat/net/reports/AbstractReportGeneratorService.java

```diff
@@ -72,13 +72,14 @@ protected AbstractReportGeneratorService(
 
     @Override
     public final CompletableFuture<Path> exec(
-            RecordingDescriptor recordingDescriptor, String filter) throws Exception {
+            RecordingDescriptor recordingDescriptor, String filter, String acceptHeader)
```

Comment:
Naming this parameter as `acceptHeader` seems off. This class is "deeper inside" Cryostat than an HTTP handler, so it doesn't and shouldn't "know" anything about HTTP or `Accept` or any other header. Using the `HttpMimeType` for this purpose therefore also seems off, though at least the enum would provide some more type information than a `String` does.

Since there are really only two accepted values for this parameter (HTML or JSON), I think it would be better to implement this either by having two methods (`execHtml` and `execJson`, or better to avoid those specific names and use say `execFormatted` and `execRaw`), or do `exec(RecordingDescriptor recordingDesciptor, String filter, boolean formatted)`, or something else similar.

Maybe even better, but much more work and thinking required, would be something like this:

```java
CompletableFuture<Report> exec(RecordingDescriptor recordingDescriptor, String filter);

Report report = exec(recording, someFilters).get();
String json = report.asJsonString();
String html = report.asHtml();
```